include/rocksdb/env.h

```diff
@@ -385,6 +385,35 @@ class Env : public Customizable {
     return Status::NotSupported("LinkFile is not supported for this Env");
   }
 
+  // Open Sync and Close the file to flush the file content to file system.
+  // See FileSystem::SyncFile comment for details
+  //
+  // There are 2 reasons we need this API. Both of them are for tests.
+  //
+  // 1. The default implementation of SyncFile API is essentially a wrapper of
+  // other FileSystem APIs. FaultInjectionTestEnv uses this default
+  // implementation to call other FileSystem APIs defined at
+  // FaultInjectionTestEnv class to inject failurses. See
+  // FaultInjectionTestEnv::SyncFile for more details
+  //
+  // 2. Some of old tests are using LegacyFileSystemWrapper.
+  // LegacyFileSystemWrapper forwards the API call to EnvWrapper, which forwards
+  // to CompositeEnv, and then forwards to the actual FileSystem implemention.
+  // Without this API in Env, LegacyFileSystemWrapper will not be able to
+  // forward the API call to EnvWrapper, causing the default FileSystem API to
+  // be called.
+  //
+  // Due to the above reason, adding a new API in FileSystem, would very likely
+  // require the same API to be added to Env.
```

Comment:
This is a good essay on why the function exists in Env, but I think it's distracting and irrelevant to the general public API user. Perhaps it can go in the default implementation.